### PR TITLE
feat: propagates request related error into their corresponding call stack

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

**Comment by cesarenaldi** [comment]
I am using a different approach than originally planned. I created a dedicated interface `IGraphQLClient` that will be used by gateways and presenters that needs to make API calls. Then I created an extended version of the `ApolloClient` (I called `LensApolloClient` for lack of better ideas) which implements this interface.

I worked around the `ApolloClient.watchQuery` returning a special type of Zen Observable (called `ObservableQuery`) by creating a `poll` method that returns a plain Zen Observable associated with an Apollo ObservableQuery that is already polling the given query.

This will replace the use of `ApolloClient.watchQuery` in places like `TransactionObserver` (see https://github.com/lens-protocol/lens-sdk/blob/main/packages/react/src/transactions/infrastructure/TransactionObserver.ts#L125).

**Review by desfero — APPROVED** [review]

File: packages/api-bindings/src/apollo/LensApolloClient.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  assertError(error);
+
+  if (isGraphQLValidationError(error)) {
+    return new ValidationError(error);
+  }
+  return new UnspecifiedError(error);
+}
+
+export class LensApolloClient<TCacheShape extends NormalizedCacheObject = NormalizedCacheObject>
+  extends ApolloClient<TCacheShape>
```
[the remark this answers is not in the document]
Comment (cesarenaldi) [reply packages/api-bindings/src/apollo/LensApolloClient.ts:41]: This extended version of `ApolloClient` is the same injected in the render tree via the `LensProvider`. This way also our read hooks use this instance and have normalized error types for free.

File: packages/react/src/publication/__tests__/useReaction.spec.ts [hunk 2]
```diff
@@ -43,8 +43,6 @@ describe(`Given the ${useReaction.name} hook`, () => {
           reactionType: ReactionType.UPVOTE,
         });
       });
-
-      expect(result.current.error).toBe(null);
```
[the remark this answers is not in the document]
Comment (cesarenaldi) [reply packages/react/src/publication/__tests__/useReaction.spec.ts:47]: yeah I agree it looks odd. This test relies on the mock to not throw an error.

Although it's possible to have test doubles for which we write expectations before the exercise and the verification phase is about checking if those expectations has been met, I give it to you it's not very obvious here.

I'll add an expectation that although not 100% the point of the test is accurate.

**Review by krzysu — APPROVED** [review]

File: packages/domain/src/use-cases/publications/HidePublication.ts [hunk 3]
```diff
@@ -1,11 +1,9 @@
-import { PromiseResult } from '@lens-protocol/shared-kernel';
-
 export type HidePublicationRequest = {
   publicationId: string;
 };
 
 export interface IHidePublicationGateway {
-  hide(request: HidePublicationRequest): PromiseResult<void, never>;
```
Comment (cesarenaldi) [inline packages/domain/src/use-cases/publications/HidePublication.ts:8]: This was unnecessary.

**Pull request merged**